pronto_mia/lib/core/factories/error_message_factory.dart

```diff
@@ -1,6 +1,92 @@
-// TODO: Implement error distinction
+import 'package:graphql/client.dart';
+
+// ignore: avoid_classes_with_only_static_members
 class ErrorMessageFactory {
-  String getErrorMessage(dynamic error) {
-    return 'Es ist ein unerwarteter Fehler aufgetreten.';
+  static const String _unknownError =
+      'Es ist ein unerwarteter Fehler aufgetreten.';
+  static const String _networkError =
+      'Es konnte keine Verbindung zum Server aufgebaut werden.';
+
+  static String getErrorMessage(dynamic error) {
+    String message = _unknownError;
+
+    if (error is OperationException) {
+      message = _getGraphQlErrorMessage(error);
+    }
+
+    return message;
+  }
+
+  static String _getGraphQlErrorMessage(OperationException error) {
+    if (!_isNetworkAvailable(error)) {
+      return _networkError;
+    }
+
+    final serverException = error.linkException as ServerException;
+
+    if (serverException.parsedResponse == null ||
+        serverException.parsedResponse.errors == null ||
+        serverException.parsedResponse.errors.isEmpty ||
+        serverException.parsedResponse.errors.length > 1) {
+      return _unknownError;

```

Comment:
Is it necessary to both include `serverException.parsedResponse.errors.isEmpty` and `serverException.parsedResponse.errors.length > 1`

Reply:
It is not strictly necessary. But if more than one error is received we cannot safely assume that the first one is the one we need.